### DISCUSSION TITLE
.NET 8.0 - Oqtane 5.0.0 Issues To Test

### DISCUSSION THREAD

**Discussion by thabaum** [oqtane/oqtane.framework#3390] [Q&A]
> The next release will be Oqtane 5.0 and will be based on .NET 8 (LTS). The goal is to release Oqtane 5.0 on ~Nov 15 (immediately after .NET 8 officially ships). Oqtane 5.0 will be focused on .NET 8 compatibility and regression testing of all current scenarios. It may also embrace some of the new Blazor rendering modes in .NET 8, however this will be dependent on the outcome of research to be conducted over the coming month.

I have successfully upgraded the Oqtane Framework entire solution referencing past upgrades for I believe 100% of the files in a current fork I am working on.  I can upload the changes to a repo on my account to save anyone the hassle if desired.

I cannot test all environments such as Azure and all the different database connections, I don't currently have an interest in it either.

I have tested the following

- installing fresh version from running Server mode
- Running from IIS Express after server install
- navigation of pages including admin and new pager feature in pages admin module
- logging in
- adding new page
- add new html module to page
- add existing module to page
- publish page
- add admin module to page
- creating new module template
- building new module .net 8
- adding new module to page
- uploading site logo
- saving site settings
- Add/delete/edit user
- add/edit/move/delete modules on page
- Set url in MauiConstants.cs and was able to load Windows Machine after adding the Maui project to the solution.
- Create Sites
- Manage Sites
- Edit Sites
- Server Hosting Mode Sites Work
- WebAssembly Hosting Mode Sites: **Blank Theme Or Page Rendering Issue ?! Maybe my upgraded Oqtane issue or .NET 8 issue**
- Hybrid Hosting Mode - Site is disabled error message similar to 4.0.6 ( expected ?)

This is all from within visual studio running server mode.

Client project did not load a page which I believe it is due to needing to be hosted in IIS and not ran from within Visual Studio so testing here is needed I will work out and update this post as I go through things.

Not sure there is much to worry about as things seem to be working great so far in every place I have tested.  I will go through the "breaking changes" logs first to see if I can identify any issues there within the code base.  Then I plan to start testing out the new features in .NET 8.

@sbwalker is there anything you would like for me to test further?  I am probably going to start pushing some mobile development tests with Maui next.  I am going to work on iOS, Chrome OS, Windows Desktop supported systems to ensure my future application is going to do it's job both native and hybrid I am hoping.

Again I updated everything to .NET 8 and am running Preview latest RC version.  

I am going to work on the router next to see how it works with the new rendering features, see what breaks or acts funny... and maybe get a solution for anything not working properly with Oqtane Router.  This is my interest currently to push forward on so I can be sure any module components I create harness everything Oqtane can with or without these new render features in .NET 8.  

Just thought I would share as I am 100% going this direction and going to "lighten" up Oqtane as much as I can in the near future to remove anything I dont need for another project I want to build soon ripping this thing down to an extremely bare bone framework.

I would be happy to do some other tests in areas you feel maybe trouble.

I see some minor things to fix like in the new pager search, nothing functional more cosmetic or visual but not sure if I can or should push out a PR that would be acceptable until after .NET 8 migration testing finishes.

I am trying to break it any way possible.

**Comment by sbwalker** [comment]
I would suggest the main focus areas in .NET 8 would be the hosting models (ensuring the framework works fine on Server, WebAssembly, and Hybrid), multi-tenancy (creating various sites using shared and isolated tenancy with different hosting models), upgrades of existing Oqtane insrallations, custom module compatibility testing, etc...

**Reply by thabaum** [reply]
Server/Hybrid seem to work ok, I have not been able to load WebAssembly i get a blank page but this should be hosted in IIS if I recall for testing and not from within Visual Studio.

I will test the tenancy tomorrow.  Here are some updates:

I currently have tested the module creator and theme creator.

Issue I have ran into and maybe you can recall seeing in the past is application crashing when restarting the application within System Info admin panel page.

Unfortunately I did not do this first.  So I will create a fresh repo with all my changes for .NET 8 to be sure I have made them all consistently when trial/error.

But I will try to reproduce this error to see if it don't happen right out of the box:

I created 2 modules, I think the restart issue may have happened during this phase as well but I was able to get them to load and use them after starting the server and hybrid applications.

I then created a theme, and the theme is not showing up in the list, and the application restarting causing an error.

I was trying to find the notes on what "not" to do when building the theme/module first time to then hit restart.  Such as building the server project I believe is the issue.  We should put a note of this in the notification message as a reminder to me :)

I will create the new repo with .NET 8 tomorrow following my notes and make sure I go over all the files again to practice the upgrade of the code base.

The modules seem to work after re-launching the framework, but I  have not seen the theme work yet which I think maybe creating the crash event.

I will test it first on 4.0.6 tomorrow and be sure it is not something I am doing wrong or part of some other issue not related to .NET 8 as well.

**Reply by sbwalker** [reply]
@thabaum in regards to "application crashing when restarting the application within System Info admin panel page" - this has always been the case when running Oqtane from within Visual Studio - it is not an issue. When you create modules or themes from within Oqtane you need to compile them and then restart the framework for it to be aware of them.

**Reply by sbwalker** [reply]
@thabaum Both Blazor WebAssembly and Blazor Hybrid can be used in Visual Studio. For Blazor WebAssembly you can add a new Site to an existing installation and choose the Hosting Model as WebAssembly. I would also recommend you watch https://www.youtube.com/watch?v=A7KnkHe3AXY to see info on running Oqtane on Blazor Hybrid.

**Reply by thabaum** [reply]
@sbwalker  

[https://www.youtube.com/watch?v=xyus7zlmgdc](https://www.youtube.com/watch?v=xyus7zlmgdc)

I am watching this one as well right now and it cleared some things up for me as I just got back to my office.  I believe I have ran the "Hybrid" approach but simply by running a server instance of Oqtane in visual studio and fire up a second instance of visual studio with Maui running while the server is still running.  I even added the Maui project to the Oqtane project and I can run both.  I just cant seem to run the "Client" version although it fires up while the server is running and runs on a different port.  It works in all my other projects like this I have created, but I dont think I ever got it working in Oqtane this way, I had to create a second site like you said if I recall and did it using IIS pointing to the release files.

The issue I have is not being able to run the WebAssembly Client Project at the same time as Server Project (two instances of visual studio) as I tried both ways except I believe the one you outlined which is create a second site using the same database as the server database.  I was making a WebAssembly site without a site running on the database that has a site running in server mode.  I was running a client project and server project at the same time if this makes sense.  It works with Maui project and Server project (two instances of visual studio) but I guess not with WebAssembly (Client) unless you create a second site within Oqtane?!  So this is what I will try now.

I will make some notes, get the process down and submit some docs even if it dont get accepted I can reference my PR. I will fully review this other video to see what I am doing wrong as I did create a second site instance on a second database instance which maybe the issue as I was trying to run a full WebAssembly version on its own without a server version running in the end.  I did try running the second version of "Client" project but I think I have the process down wrong compared to what I should be doing to make a WebAssembly or client project version run.

It is just all other projects I have created generally have a client project (WebAssembly) and a server project (Server) and a native project (Maui) along with a "Shared" project for code to share between the projects.  Oqtane follows a similar structure but has a little different setup to make work properly.  So I am constantly fighting between what I have learned heavily over the past year and what I have to do here.  I deleted most all I had to start fresh with a more clear understanding.

Once I get a full understanding again as I had this down before I will provide some type of documentation response so I don't ask again as I feel bad having hit you up with these types of questions and I really don't want to wait for a response if I can help it.

Thanks for the reply and taking on some tough feedback that maybe taken out of context if not taken with a grain of salt as I mean nothing but the best for this community.

I will be focusing on this all week long, please forgive me if I am a bit more active than normal.  I want to get a "Testing" setup going to help and maybe explore building some unit tests once I get that down as a side project that maybe I can share to help down the road.

**Reply by thabaum** [reply]
@sbwalker Here is my summary after testing many scenarios and going through the process of understanding how to setup sites in Oqtane.  I did run into an issue with .NET 8 where WebAssembly projects didnt fully load up the theme.  The sites loaded but the theme and components all blank except the background color.  Blazor Server works, Maui works, Hybrid "Site is Disabled" works same as 4.0.6 but I am not sure that is expected behavior.  Sites only work in visual studio if you use the same port as the master default site managing from within the Oqtane framework which could be a firewall/visual studio issue as Visual Studio may not open that port up to take requests.  Below was my workflow along with my snags and things I had for notes along the way.

Without having to read all below is there anything that you have seen before where switching a site to WebAssembly or creating a site with WebAssembly has an issue as described?  I created databases, sites and so on, but again the port had to be the same was and each had to have it's own alias.  From other projects in Maui this is what got me a little confused and since I do not set these up often enough to remember the does and donts in Visual Studio and Oqtane this cleared up a lot for me.

First I tested with fresh 4.0.6 as I can see the databases being created, but I cannot access the sites.  I can access the original site in server mode, I can access the original master site from .NET MAUI editing the `MauiConstants.cs `file.  I cannot seem to access any other sites to browse or edit even while the server version is running with a second instance of VS or not.  The databases are created but when I look to select a database it shows the site name and not the actual database name, which maybe is OK as it lets you know which site you want the new site to share a database with by name, but it doesn't let you know which database you are connecting to specifically.  In other words I have many Oqtane-2023.... databases created, which goes to which site I have no clue without digging into the master database or making note of it.  When I create sites I will name them more specifically next test run.

I watched both those videos a few times now and seen them all when they played live I believe.  I have been through the slides and seen the basics work.

I went through with details below along with results from my workflow so you can tell me what is expected/unexpected behavior so I can ignore these as issues and spot ones that maybe issues.  I will try to do my best to work a number of scenarios out as I am trying to break this in any way and make it work in any way utilizing Visuat Studio only to start.  Help me here I can help with docs.

Just imagine I am very green at this so bare with me.  You can just put down a number for the test scenario and say "expected" and maybe a reason why so I can explain it later in an Oqtane docs PR.  I need this for future reference and I don't wish to bug you again if I can help it relating to this matter.  A lot of this I expect to be expected but I just want to run through it so I can get all the fine details covered for all potential trial and error use cases to have for a reference you can put a link here to.

Fresh 4.0.6 dev branch 10-17-2023
Nothing Installed Tests:

1. Launching the Oqtane Framework "Client" project  "Client" (dotnet.exe) or "IISExpress" in Visual Studio Debug or Release Mode a browser tab opens with http://localhost:44358/ as a blank page.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/4374500b-a2b8-43aa-bc5a-557f8e8fc5c7)

2. Launching the Oqtane Framework "Client" project  "Client" (dotnet.exe) or "IISExpress" in Visual Studio Debug or Release Mode a browser tab opens with http://localhost:44357/ and a setup database configuration and application page is shown.

I configure a database and host user as instructed.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/8aa1a3c0-1e86-47da-8c78-d0c191811ec8)

Site and database is created.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/18d29759-1ddc-4563-b921-ad5ebe19ef5a)

3. Set the .NET `MauiConstants.cs` in the Oqtane Maui Project to use `http://localhost:44357/` and open the Maui project in another Visual Studio instance either by opening the same Oqtane Solution and adding the Maui project to current Oqtane Solution "Add Project" and while the "Server" version is running on one instance of Visual Studio using that same `http://localhost:44357/` open a second instance of the Oqtane solution and run the Maui project which will allow you to run the Windows App Store application version simultaneously as it will use the "Server" instance of Visual Studio for the node to connect to the API controllers connected to the data repositories.

The .NET Maui project can be opened without adding to the Oqtane server solution, I just do that as this is how all my other Blazor/Maui solutions are.  It works and I am used to it so why not?  If an issue I would like to know.  But this works great as far as I can tell.

4. With the server running on `http://localhost:44357/` I would be thinking the "Client" or "WebAssembly" version would also need the "Server" version running and connect to its API via WebAssembly or "Client" Project or Site Instance.  Problem is it wants to connect via http://localhost:44358/ and no api node is running for it to connect to so maybe I am missing a step here?!.  I believe this can be edited in the project but not sure if needed or the right approach for Oqtane Framework.

5. All sites dont seem to work or are able to "edit" or "browse" from visual studio.  Maybe these sites need different URLS but same port?  I am going to step through these scenarios now.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/a011f48f-375b-468a-baf1-f6f80afb4a7a)
Try to use localhost:44357 with WebAssembly Site to add to the "Server Site" for example which I feel could be a use case and it is not allowed.

6. Create a second site with `oqtane.devsite:44357` or something set using "Same" port as the "Server" Master database site app and the app actually runs!  I can log in like I am using the server and from the same single instance of Visual Studio.  It loads in HTTPS mode which makes a lot of insecure requests to the original Server

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/bd3a58f8-6c46-4d57-9b33-5d6227a1aa5d)

Issue I see is that we should be able to really add a server and a WebAssembly instance and use the same site domain alias if it is using the same database and tenant to marry up with.  I have never really heard this approach discussed but makes sense it works which is why I just tested it.

I am not 100% sure how to view what all is being loaded to verify this is "WebAssembly" however this should give clue that it is in image below as I get all kinds of HTTP cryptography errors, but it works.   Since running HTTP site without a valid SSL certificate this is expected.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/476a5272-5311-4278-82a6-0ffbd038c44b)

7.  I used the following similar configuration but with "Hybrid" mode to see if it got similar results.  The results where "Site Is Disabled"
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/58ff8d2d-af4e-498d-8cbc-ee63934f2269)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/8c27c6a7-c35e-46b0-b4fe-28b95aec66a4)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/f0a852e8-fc1d-4f18-9e2b-10042548e2e8)

End results is the Server "localhost" and WebAssembly "oqtane.devsite" co-existed but "Hybrid" resulted in a disabled site using the same port as "Server" from the same instance of visual studio.

8.  Open second visual studio same Oqtane Framework 4.0.6 solution.  Both the Server runtime and the WebAssembly runtime sites can run simultaneously, and you can run your Maui app as well.

Issue is, I am expecting to be able to run both but I suppose i can see why this would not be best approach, however, it is the only way I have seen "WebAssembly"  run within visual studio.

9. Client wants to run with `http://localhost:44358/` so I want to play with this domain and port next and using the same master database so there is no setup.
Tested "Hybrid"
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/81f733cb-9ff3-481c-8d96-0c35ab87a680)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/71ac201e-17db-4ec3-84e0-8e15492b756c)
Client project does not run from solution either.
Tested "Server" with new domain same port:

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/56383c28-9526-4d9f-8b8b-915da9b87acb)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/4fa00d97-b0dc-400b-a86a-8280adc4eb1d)
No Go so wont try to test web assembly this way as it needs a server instance to run and allow it to connect to the API for database operations.

10.  Site with a new database created, server, using port 44358
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/d9ba57d7-5141-4912-937a-acf2aa02065f)

results : unable to connect

11.  connect to that same new tenant Db Blazor Server runtime but using the same port and new domain name as the 44357 addresses used previously
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/0fc4018c-5086-4d67-aee5-33deab84303f)

and we get a site that loads:

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/28caabaa-be3a-45fa-b81c-e1dfd8ac91fa)

Maybe a firewall issue with Visual Studio?

Now if I change the runtime within the site to "Blazor WebAssembly" it seems to run.

Switch to "Blazor Hybrid" and "Site is Disabled" and can no longer access it from within the framework when logged in as host.

So no luck with "Blazor Hybrid"

Some interesting findings with "WebAssembly" that may have cleared up some confusion.

I will now go through these same tests that "worked" for testing the different runtimes and creating sites.  I see the custom router in full use cases more so I can try to visualize how the .NET 8 auto render mode may work with Oqtane if it does not already out of the box...

I think an "Auto" mode for Host Model is on deck if not working out the box in .NET 8 for Oqtane.

This is as much testing as I wish to do for now on my .NET 8 upgraded solution.

I will report results when I go back and say the results are same for each scenario in .NET 8 then things are "normal" going forward in regards to these tests.

I noticed Site Settings "Is Deleted" option, would be nice to delete in Site Management for sites you created that you cannot access.

I just typed my workflow and thought process, in conclusion I feel at the end I know of scenarios that actually work now and how to setup these sites to test at least what I know works as expected.


.NET 8 Results testing (somethings maybe different and did not test last version)
1. Selecting Blank Site and Blazor Theme Blazor Server Default Port and get a blank site but seems to be working, just cant see the theme!  No way to edit the site settings without database hack.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/e46db7cb-f7f5-4723-86d1-ee4c8a8171ff)

2. Blazor Server wtih Oqtane Theme and it works as expected
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/cc2aed01-a7d7-442a-8188-8b264e5f649c)

Ok

3.  Changed to "WebAssembly" and get a blank theme, I may have not edited all the theme files for .NET 8 maybe?! or this is an issue.  

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/4a0bf2b9-143a-4d6e-b01d-ab5782974110)

4.  Created Blazor Server with a "Tenant DB" and "Oqtane Theme" "Blank Template" and worked ok.

![image](https://github.com/oqtane/oqtane.framework/assets/13577556/7f184dad-e54d-40f3-a19b-782d04388f17)
All sites show up when logging in as the host user account created in first site.

So maybe I have an issue where I still need to change some things for .NET 8 WebAssembly and the theme to load.  I could not get the Theme Creator to work and load the theme after I build it so not sure if related or just error on my part.  It wasnt very hard before to setup a new theme to create so just something to keep in mind.

Some errors in the network developer tools logs:
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/b720d362-3e0e-4574-8eae-a5d1d12792f9)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/3184d3ad-3122-40dc-bee1-9c5e804f38d6)
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/2f70380e-4033-4ec3-819c-280a9ea4bedb)

**After all that testing, I believe everything is working as expected tested so far except the WebAssembly not loading the theme which could be something in my Oqtane Framework code base upgrade to .NET 8 or a .NET issue.**

The videos are helpful, but they did not get me to launch a second site properly as I was trying different ports which was my issue for testing in Visual Studio which I have done in the past with other projects if I recall which is why I had an issue creating sites with ports outside of the default port that Oqtane Framework Solution Server Project uses.  A green VS mistake I am sure but nothing told me otherwise.

Found another issue:  If you enter the wrong Host username and password an error happens saying click on link to view log, but no log is made even though the message says to check logs to review it.  Not a big issue but something that may lead to a little confusion.  A log should be placed that says host credentials were not supplied while trying to create a site or not reference the logs since there isn't one created in the message.

**Comment by sbwalker** [comment]
@thabaum can you please post a screenshot of the System Info page in your .NET 8 installation

**Reply by thabaum** [reply]
![image](https://github.com/oqtane/oqtane.framework/assets/13577556/3e19a784-cfce-4404-963c-9e1e4be5ca6d)

**Reply by thabaum** [reply]
I upgraded VS to preview 4 today, still same results.

**Reply by sbwalker** [reply]
thank you... the memory allocation seems larger in .NET 8 - but maybe its related to the dev environment

**Reply by thabaum** [reply]
I did notice that was a bit large...

**Reply by thabaum** [reply]
I have also been trying to play with the new component render modes with limited knowledge of what exactly to do.

I was trying to turn the entire application into auto but I was only able to add one line that didnt create a build error to the Server Project > Startup.cs.

I was able to add the following without a build error only but I am not sure and doubt it really had any affect on Oqtane:
```
           services.AddRazorComponents()
            .AddInteractiveServerComponents();
```

Since I Tried the following but error in build:

```
builder.Services.AddRazorComponents()
    .AddInteractiveServerComponents()
    .AddInteractiveWebAssemblyComponents(); <-- is not found
```

`MapRazorComponents` not found when adding the following as well as instructed:

```
app.MapRazorComponents<App>()
    .AddInteractiveServerRenderMode()
    .AddInteractiveWebAssemblyRenderMode();
```

I tried adding the `@using static Microsoft.AspNetCore.Components.Web.RenderMode` to the Client Project _imports file but "RenderMode" is not found and test out some components with `<Dialog @rendermode="InteractiveAuto" />`

```
Interactive Server | @attribute [RenderModeInteractiveServer]
Interactive WebAssembly | @attribute [RenderModeInteractiveWebAssembly]
Interactive Auto | @attribute [RenderModeInteractiveAuto]
```
and then also try adding the following in an appropriate file which maybe a rendermode setting that needs to be applied via database setting when pages get created but I was going to hardwire things in to test as I could:
```
@page "/render-mode-4" <--- which gets ignored by Oqtane Router as it uses the database settings.
@rendermode RenderMode.InteractiveAuto

<button @onclick="UpdateMessage">Click me</button> @message

@code {
    private string message = "Not clicked yet.";

    private void UpdateMessage()
    {
        message = "Somebody clicked me!";
    }
}
```
I want to add auto rendering mode for the entire application was a goal first to see if I can get it to take affect in Oqtane so .NET just figures out what is best depending on client browser and network connectivity.

References I used in docs:
[https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#enable-support-for-interactive-render-modes](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#enable-support-for-interactive-render-modes)

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serverrazorcomponentsbuilderextensions.addinteractiveservercomponents?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.serverrazorcomponentsbuilderextensions.addinteractiveservercomponents?view=aspnetcore-8.0)

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.routing.iendpointroutebuilder?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.routing.iendpointroutebuilder?view=aspnetcore-8.0)

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.razorcomponentsendpointconventionbuilder?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.builder.razorcomponentsendpointconventionbuilder?view=aspnetcore-8.0)

[https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#enable-support-for-interactive-render-modes](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/render-modes?view=aspnetcore-8.0#enable-support-for-interactive-render-modes)

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.web.rendermode?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.web.rendermode?view=aspnetcore-8.0)

[https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.web.rendermodeinteractiveautoattribute?view=aspnetcore-8.0](https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.web.rendermodeinteractiveautoattribute?view=aspnetcore-8.0)

Just thought I would share I am trying to wrap my mind around how this might be able to be enhanced if possible.  Not a huge deal but something that is "hyped" up in .NET 8.

I dont see any version 8 nuget packages to install, and I know I have latest versions of .NET 8 runtimes and hosting packages installed on the system.

**Reply by thabaum** [reply]
after reviewing the new NET8 branch and from your review of the project files I had I see the issue was relating to the packages not utilizing the latest ( I had version 7 still ) so I will go through my testing once again with the new branch.

**Reply by thabaum** [reply]
[Migrations and Version Updates 7.0 to 8.0](https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0&tabs=visual-studio)

Another doc file for resource.
